spring-boot/proxy/src/test/java/eu/coatrack/proxy/security/LocalApiKeyManager_AbstractTestSetupProvider.java

```diff
@@ -25,21 +25,25 @@
 import java.util.ArrayList;
 import java.util.List;
 
+import static org.apache.commons.codec.digest.DigestUtils.sha256Hex;
 import static org.mockito.Mockito.mock;
 
 public abstract class LocalApiKeyManager_AbstractTestSetupProvider {
 
     protected ApiKey apiKey;
-    protected List<ApiKey> apiKeyList;
+    protected List<ApiKey> hashedApiKeyList;
```

Comment:
In here, it seems that the list of API keys is hashed on the Proxy side.  So if this is the case, then:
So as the list of API Keys are sent from the admin, the list should be first hashed and then sent, because if it is only hashed on the proxy side, it means that if there is hacker attack, the hacker can still get the list unhashed from admin.

Reply:
You are correct. The API key hashing shall be performed by the CoatRack Web Application in production. But this is a template for unit testing classes which require hashed API keys as mock objects.